File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/importwebresourcesvb/vb/importwebresources2.vb
' <snippetimportwebresources2>


 'Set the Web Resource properties
 Dim wr As WebResource =
     New WebResource With
     {
         .Content = getEncodedFileContents("../../" _
                                           &amp; awebResource.path),
         .DisplayName = awebResource.displayName,
         .Description = awebResource.description,
         .Name = _customizationPrefix &amp; awebResource.name,
         .LogicalName = WebResource.EntityLogicalName,
         .WebResourceType =
         New OptionSetValue(Int32.Parse(awebResource.type))
     }

 ' Using CreateRequest because we want to add an optional parameter
 Dim cr As CreateRequest = New CreateRequest With {.Target = wr}
 'Set the SolutionUniqueName optional parameter so the Web Resources will be
 ' created in the context of a specific solution.
 cr.Parameters.Add("SolutionUniqueName",
                   _ImportWebResourcesSolutionUniqueName)

 Dim cresp As CreateResponse = CType(_serviceProxy.Execute(cr), 
     CreateResponse)

' </snippetimportwebresources2>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/importwebresources/cs/importwebresources2.cs
// <snippetimportwebresources2>


 //Set the Web Resource properties
 WebResource wr = new WebResource
 {
     Content = getEncodedFileContents(@"../../" + webResource.path),
     DisplayName = webResource.displayName,
     Description = webResource.description,
     Name = _customizationPrefix + webResource.name,
     LogicalName = WebResource.EntityLogicalName,
     WebResourceType = new OptionSetValue(Int32.Parse(webResource.type))
 };

 // Using CreateRequest because we want to add an optional parameter
 CreateRequest cr = new CreateRequest
 {
     Target = wr
 };
 //Set the SolutionUniqueName optional parameter so the Web Resources will be
 // created in the context of a specific solution.
 cr.Parameters.Add("SolutionUniqueName", _ImportWebResourcesSolutionUniqueName);

 CreateResponse cresp = (CreateResponse)_serviceProxy.Execute(cr);

// </snippetimportwebresources2>